### PR TITLE
label 83

### PR DESCRIPTION
**Pull request by rpatel3001**

### CODE REVIEW

File: lib/plugins/Label_83.ts [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+    // Inmarsat C-band seems to prefix normal messages with a message number and flight number
+    let text = message.text;
+    if (text.match(/^M\d{2}A\w{6}/)) {
+        ResultFormatter.flightNumber(decodeResult, message.text.substring(4, 10).replace(/0/g, ""));
+        text = text.substring(10);
+    }
+
+    decodeResult.decoded = true;
+    if (text.substring(0, 10) === "4DH3 ETAT2") {
```
Comment (kevinelliott) [inline lib/plugins/Label_83.ts:29]: Might be helpful later if we had a comment here about this match.

**Review by kevinelliott — APPROVED** [review]
Thanks @rpatel3001 ... This over all looks good. A link to an Airframes message could be useful (in addition to that adsbx link), but not needed to merge. Unless @makrsmark has any objections, this is ready.

File: lib/plugins/Label_83.test.ts [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+  expect(decodeResult.formatted.items[0].type).toBe('origin');
+  expect(decodeResult.formatted.items[0].value).toBe('KLAX');
+  expect(decodeResult.formatted.items[1].type).toBe('destination');
+  expect(decodeResult.formatted.items[1].value).toBe('KEWR');
+  expect(decodeResult.formatted.items[2].type).toBe('aircraft_position');
+  expect(decodeResult.formatted.items[2].value).toBe('40.530 N, 74.470 W');
+  expect(decodeResult.formatted.items[3].type).toBe('altitude');
+  expect(decodeResult.formatted.items[3].value).toBe('3836 feet');
+  expect(decodeResult.formatted.items[4].type).toBe('aircraft_groundspeed');
+  expect(decodeResult.formatted.items[4].value).toBe('212');
```
Comment (makrsmark) [inline lib/plugins/Label_83.test.ts:45]: This is probably my fault, but can we get a unit on the formatted groundspeed? (Knots?)

File: lib/plugins/Label_83.test.ts [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+  expect(decodeResult.formatted.items[5].type).toBe('heading');
+  expect(decodeResult.formatted.items[5].value).toBe('-107.6');
+});
+
+test('decodes Label 83 variant 2', () => {
+  const decoder = new MessageDecoder();
+  const decoderPlugin = new Label_83(decoder);
+
+  // https://globe.adsbexchange.com/?icao=478F43&showTrace=2024-09-22&timestamp=1727022863
+  const text = '4DH3 ETAT2  0907/22 ENGM/KEWR .LN-RKO\r\n/ETA 1641';
```
Comment (makrsmark) [inline lib/plugins/Label_83.test.ts:94]: It can wait, but do you think we can pull out the tail ? `LN-RK0`

File: lib/plugins/Label_83.ts (hunk 1 above)
Comment (makrsmark) [reply lib/plugins/Label_83.ts:29]: +1 just a comment linking this to the variant in the tests

File: lib/plugins/Label_83.ts [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+        const fields = text.split(/\s+/);
+        if (fields[2].length > 5) {
+            decodeResult.raw.day_of_month = fields[2].substring(5);
+        }
+        decodeResult.remaining.text = fields[2].substring(0, 4);
+        const subfields = fields[3].split("/");
+        ResultFormatter.departureAirport(decodeResult, subfields[0]);
+        ResultFormatter.arrivalAirport(decodeResult, subfields[1]);
+        ResultFormatter.eta(decodeResult, fields[6] + "00");
+    } else if (text.substring(0, 5) === "001PR") {
```
Comment (makrsmark) [inline lib/plugins/Label_83.ts:39]: Same comment about adding a comment to the variant

**Review by makrsmark — APPROVED** [review]
Nothing blocking. Approving and will merge later today to give you time to decide whether you want to incorporate minor feedback

**Comment by rpatel3001** [comment]
updated for feedback.

i think a message about an airframes link disappeared but tbh an adsbx link is more useful for checking accuracy, no? can only pull it easily if a message comes from my local acars2pos logs though

**Comment by rpatel3001** [comment]
whoops, rebased to fix conflicts (and the airframes link message is back? github on the fritz)

**Pull request merged**